### PR TITLE
Warn about third-party JVM agents in docs

### PR DESCRIPTION
**Pull request by williamrandolph**
We've had reports of cases where Elasticsearch nodes froze mysteriously or had difficult-to-diagnose performance problems, and the eventual culprit was some monitoring agent that attached into to the JVM and interfered with what Elasticsearch needed to do. We can add a warning about this to our docs so that we have something to reference when this problem comes up again.

### CODE REVIEW

File: docs/reference/setup/install.asciidoc [hunk 1]
```diff
@@ -87,6 +87,10 @@ bundled JVM is treated as an integral part of {es}, which means that Elastic
 takes responsibility for keeping it up to date. Security issues and bugs within
 the bundled JVM are treated as if they were within {es} itself.
 
+You should not use long-running third-party agents that attach to the JVM
```
Comment (rjernst) [inline docs/reference/setup/install.asciidoc:90]: Remove "long-running"? Any java agent, regardless of how long lived, should not be attached to Elasticsearch.

**Review by rjernst — COMMENTED** [review]
Maybe this paragraph should be in its own section? I think it may get lost underneath JVM, and it isn't really about selection of the jvm as the other documentation here is about.

File: docs/reference/setup/install.asciidoc [hunk 2]
```diff
@@ -87,6 +87,10 @@ bundled JVM is treated as an integral part of {es}, which means that Elastic
...
+for deep introspection and instrumentation. Such software has been observed
+to reduce Elasticsearch performance, or even freeze or crash nodes.
```
Comment (DaveCTurner) [inline docs/reference/setup/install.asciidoc:92]: I think we should have fewer qualifiers here. It's not just _long-running agents for instrumentation_ that we want to exclude, running with any kind of agent attached is unsupported.

Also I'd prefer stronger wording than "you should not" - ideally "do not" or something equally imperative.

**Review by rjernst — APPROVED** [review]
LGTM

**Review by DaveCTurner — APPROVED** [review]
LGTM2

**Comment by williamrandolph** [comment]
Thanks! I'll merge once @jrodewig can approve it.

File: docs/reference/setup/install.asciidoc [hunk 3]
```diff
@@ -90,6 +90,14 @@ the bundled JVM are treated as if they were within {es} itself.
 The bundled JVM is located within the `jdk` subdirectory of the {es} home
 directory. You may remove this directory if using your own JVM.
 
+[discrete]
+[[jvm-agents]]
+=== JVM and Java Agents
```
Comment (jrodewig) [inline docs/reference/setup/install.asciidoc:95]:
```suggestion
=== JVM and Java agents
```

File: docs/reference/setup/install.asciidoc [hunk 4]
```diff
@@ -90,6 +90,14 @@ the bundled JVM are treated as if they were within {es} itself.
...
+
+Do not use third-party Java agents that attach to the JVM. Such software has
```
Comment (jrodewig) [inline docs/reference/setup/install.asciidoc:97]:
```suggestion
Don't use third-party Java agents that attach to the JVM. These agents
```

File: docs/reference/setup/install.asciidoc [hunk 5]
```diff
@@ -90,6 +90,14 @@ the bundled JVM are treated as if they were within {es} itself.
...
+been observed to reduce {es} performance, or even freeze or crash nodes.
```
Comment (jrodewig) [inline docs/reference/setup/install.asciidoc:98]:
```suggestion
can reduce {es} performance, including freezing or crashing nodes.
```

**Review by jrodewig — APPROVED** [review]
Approved. I left some suggestions. Please accept the fix to the heading case. You can take or leave the rest as you see fit. Thanks!

**Pull request merged**